src/table.rs

```diff
@@ -296,12 +318,16 @@ impl<T: Slot> TablePage for Page<T> {
         std::any::type_name::<Self>()
     }
 
-    unsafe fn memos(&self, slot: SlotIndex, current_revision: Revision) -> &MemoTable {
-        self.get(slot).memos(current_revision)
+    unsafe fn memos(&self, slot: SlotIndex, current_revision: Revision) -> MemoTableWithTypes<'_> {
+        self.memo_types
+            .attach_memos(self.get(slot).memos(current_revision))
     }
 
-    fn memos_mut(&mut self, slot: SlotIndex) -> &mut MemoTable {
-        self.get_mut(slot).memos_mut()
+    fn memos_mut(&mut self, slot: SlotIndex) -> MemoTableWithTypes<'_> {
+        // SAFETY: We have a `&mut` reference.
+        let memos = unsafe { self.get(slot).memos_no_revision() };
```

Comment:
This change seems unnecessary to me, we can just use `get_mut` and keep the `memos_mut` accessor as before

Reply:
But we need to attach the types to it anyway. I can change create a new type `MemoTableWithTypesMut`, but this just seems unnecessary. And given it does not return a mutable reference anymore, `memos_mut()` seems inappropriate name now.

Reply:
I don't mind the name change, I was more raising this regarding the unnecessary unsafe block. There is no reason to decay the mutable references in this call chain to an immutable one, doing so results in this extra unsafe that isn't really necessary otherwise

Reply:
Ah, given your other comment to my review I guess that would require duplicating the function for mutability

Reply:
What do you mean by

> duplicating the function for mutability

?

Reply:
> Problems with the borrow checker when calling `Page::get()` borrows the entire page then we cannot access `Page::memo_types`.

I interpreted this as you'd needing to keep the current version for the borrowck issue either way

Reply:
No, what I meant is that if I this function takes `&mut self`, then `Page.get_mut().memos_mut()` clashes with `Page.memo_types`, leading to a borrowck error.